src/dbnode/integration/query_limit_test.go

```diff
@@ -0,0 +1,107 @@
+// +build integration
+
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/m3db/m3/src/dbnode/client"
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/storage"
+	"github.com/m3db/m3/src/dbnode/storage/index"
+	"github.com/m3db/m3/src/dbnode/storage/limits"
+	"github.com/m3db/m3/src/m3ninx/idx"
+	"github.com/m3db/m3/src/x/ident"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+func TestQueryLimitExceededError(t *testing.T) {
+	testOpts, ns := newTestOptionsWithIndexedNamespace(t)
+	testSetup := newTestSetupWithQueryLimits(t, testOpts)
+	defer testSetup.Close()
+
+	require.NoError(t, testSetup.StartServer())
+	defer func() {
+		require.NoError(t, testSetup.StopServer())
+	}()
+
+	var (
+		nowFn     = testSetup.StorageOpts().ClockOptions().NowFn()
+		end       = nowFn()
```

Comment:
Nit: this would give you cleaner values in case the test fails (or you are debugging):
```suggestion
		end       = nowFn().Truncate(time.Hour)
```

Reply:
https://github.com/m3db/m3/pull/3030/commits/2ad45b7db3180d473abfeaff864c79b68b6661b3#diff-b6f7fb2a6e9bec6f25fac6ebeea5dc48cebfeeee31918b6956c634d25eec29ffR54